aten/src/ATen/autocast_mode.h

```diff
@@ -70,7 +70,8 @@ inline at::ScalarType promote_type(at::ScalarType current, Arg0 arg0, Args... ar
 Logic to apply cached casting to any Tensor argument.
 ****************************************************/
 inline bool is_eligible(const Tensor& arg) {
-  return (arg.defined() && arg.is_cuda() && arg.is_floating_point() && (arg.scalar_type() != at::kDouble));
+  return (arg.defined() && (arg.is_cuda() || arg.is_xla()) && arg.is_floating_point() &&
```

Comment:
This seems fine but you'll need to rebase or remerge master to pick up https://github.com/pytorch/pytorch/commit/22c3ae8b576674e52e1972abe13891aaf4e99089.

Reply:
Done.